src/general/LimitOrderHook.sol

```diff
@@ -0,0 +1,797 @@
+
+            // set the epoch as filled
+            epochInfo.filled = true;
+
+            uint128 amount0;
+            uint128 amount1;
+
+            // modify the liquidity to remove the epoch liquidity from the pool
+            (BalanceDelta delta,) = poolManager.modifyLiquidity(
+                key,
+                IPoolManager.ModifyLiquidityParams({
+                    tickLower: tickLower,
+                    tickUpper: tickLower + key.tickSpacing,
+                    liquidityDelta: -int256(uint256(epochInfo.liquidityTotal)),
+                    salt: 0
+                }),
+                ZERO_BYTES
+            );
+
+            // if the amount of currency0 is positive, mint the currency0 to the hook
+            if (delta.amount0() > 0) {
+                poolManager.mint(address(this), key.currency0.toId(), amount0 = uint128(delta.amount0()));
+            }
+
+            // if the amount of currency1 is positive, mint the currency1 to the hook
+            if (delta.amount1() > 0) {
+                poolManager.mint(address(this), key.currency1.toId(), amount1 = uint128(delta.amount1()));
+            }
+
+            // add the amount of currency0 and currency1 to the epoch info
+            unchecked {
+                epochInfo.currency0Total += amount0;
+                epochInfo.currency1Total += amount1;
+            }
+
+            // set the epoch as default (inactive)
+            setEpoch(key, tickLower, zeroForOne, EPOCH_DEFAULT);
+
+            // emit the fill event
+            emit Fill(epoch, key, tickLower, zeroForOne);
+        }
+    }
+
+    /**
+     * @dev Get the crossed ticks for a given pool after a price change.
+     *
+     * @param poolId The pool id.
+     * @param tickSpacing The tick spacing.
+     * @return tickLower The lower tick.
+     * @return lower The lower tick.
+     * @return upper The upper tick.
+     */
+    function _getCrossedTicks(PoolId poolId, int24 tickSpacing)
+        internal
+        view
+        returns (int24 tickLower, int24 lower, int24 upper)
+    {
+        tickLower = getTickLower(getTick(poolId), tickSpacing);
+        int24 tickLowerLast = getTickLowerLast(poolId);
+
+        if (tickLower < tickLowerLast) {
+            lower = tickLower + tickSpacing;
+            upper = tickLowerLast;
+        } else {
+            lower = tickLowerLast;
+            upper = tickLower - tickSpacing;
+        }
+    }
+
+    /**
+     * @dev Get the last tick lower.
+     *
+     * @param poolId The pool id.
+     * @return tickLowerLast The last tick lower.
+     */
+    function getTickLowerLast(PoolId poolId) public view returns (int24) {
+        return tickLowerLasts[poolId];
+    }
+
+    /**
+     * @dev Set the last tick lower.
+     *
+     * @param poolId The pool id.
+     * @param tickLower The tick lower.
+     */
+    function setTickLowerLast(PoolId poolId, int24 tickLower) private {
+        tickLowerLasts[poolId] = tickLower;
+    }
```

Comment:
small nit and personal preference -- i prefer in-lining the storage write as `tickLowerLasts[poolId] = tickLower`, especially since were only writing storage in two logic flows